src/hazelcore/Statics_Exp.re

```diff
@@ -1203,19 +1316,27 @@ and ana_fix_holes_operand =
       };
     }
   | Case(_, scrut, rules) =>
-    let (scrut, scrut_ty, u_gen) =
+    let (scrut, ty_scrut, u_gen) =
       syn_fix_holes(ctx, u_gen, ~renumber_empty_holes, scrut);
+    let (_, _, _, common_type) =
+      syn_fix_holes_rules(ctx, u_gen, ~renumber_empty_holes, rules, ty_scrut);
```

Comment:
I was expecting an ana_fix_holes followed by a call to syn_rule_types to determine whether a common type exists or not -- not sure we need to do a full syn_fix_holes here